core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3145,7 +3149,8 @@ public PhysicalOperation visitRefreshMaterializedView(RefreshMaterializedViewNod
         public PhysicalOperation visitTableWriter(TableWriterNode node, LocalExecutionPlanContext context)
         {
             // Set table writer count
-            int writerCount = isLocalScaledWriterExchange(node.getSource()) ? getTaskScaleWritersMaxWriterCount(session) : getTaskWriterCount(session);
+            Optional<ExchangeNode> scaleWriterExchangeNode = getScaleWriterExchangeNode(session, node.getSource());
+            int writerCount = scaleWriterExchangeNode.isPresent() ? getTaskScaleWritersMaxWriterCount(session) : getTaskWriterCount(session);
```

Comment:
Is this general rule that we are using Optional `isPresent` with ternary instead of we use map/orElseGet?